webchat/utils/chatClient.js

```diff
@@ -0,0 +1,328 @@
+'use strict';
+(function () {
+  const app = angular.module('webchat');
+
+  app.factory('ChatClient', ['Chat', 'WEBSOCKET', (Chat, WEBSOCKET) => {
+    const ChatClient = class ChatClient {
+      /**
+       * Create a new ChatClient (WebRTC wrapper),
+       * and automatically start a WebSocket connection.
+       * The ChatClient also mediates connections by Chat objects
+       * before a RPC connection is succesfully completed.
+       * @param {string} id - firebase id of the user
+       */
+      constructor(id) {
+        this.id = id;
+        this.ws = {};
+        this.retries = 0;
+        this.eventHandlers = {};
+        this.chats = {};
+        this.users = {};
+        this.stream = undefined;
+        this.startWebsocket();
+      }
+
+      /**
+       * Starts a new connection with the websocket server.
+       */
+      startWebsocket() {
+        const websocket = `ws://${WEBSOCKET.hostname}:${WEBSOCKET.port}`;
+        this.ws = new WebSocket(websocket);
+
+        // Sends a signin message to the websocket as soon as the websocket connects
+        this.ws.onopen = () => {
+          this.sendToWebsocket('signin', '', {});
```

Comment:
Create method for direct connections to Websocket (without 'to' parameter), to make easier future communications.